### PR TITLE
chore(menu-toggle): updated components to use menu toggle

### LINKED ISSUE #6207: Replace uses of Dropdown (deprecated) with MenuToggle
**Describe the enhancement or change**
We've deprecated Dropdown, but it's used many places in core. This issue should replace any uses of Dropdown with Menu instead

Known places:
- [x] Breadcrumb
- [x] Card
- [x] Dual list selector
- [x] Hint
- [x] Inline edit
- [x] Log viewer
- [x] Notification drawer
- [ ] Table
- [ ] Tree view
- [ ] some additional demos

Hint: search .md and .hbs for `{{> dropdown` to find instances

### PR DESCRIPTION
**Pull request by mattnolting**
closes #6207 

~~[BackstopJS Report.pdf](https://github.com/patternfly/patternfly/files/15204083/BackstopJS.Report.pdf)~~

~~[Link to the backstop report](https://drive.google.com/file/d/1cAAATR1BN_BWpiuPP81CH8DqbAfP4Kor/view?usp=sharing)~~

@mcoker [Link to the backstop report](https://drive.google.com/file/d/17Y43_hMtETkYvW-h9FX73iz5FcYBtLL5/view?usp=drive_link)

### CODE REVIEW

**Review by srambach — COMMENTED** [review]
The menu on a menu toggle is showing up to the right 
image

There seems to be a problem if the content is too wide?
image

Various places I see the toggle scrunched up
image

File: src/patternfly/components/Toolbar/toolbar-overflow-menu-example.hbs [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {{#> toolbar-item toolbar-item--modifier=(concat 'pf-m-overflow-menu ' toolbar-overflow-menu-example-toolbar-item--modifier)}}
-  {{#> overflow-menu overflow-menu-button--aria-label="Expand button group overflow menu" overflow-menu--id=(concat toolbar--id '-overflow-menu') dropdown--IsExpanded=toolbar-overflow-menu-example--IsExpanded overflow-menu--modifier=toolbar-demo-overflow-menu--modifier overflow-menu--attribute=toolbar-demo-overflow-menu--attribute}}
-    
+  {{#>overflow-menu
+      overflow-menu-button--aria-label="Expand button group overflow menu"
+      overflow-menu--id=(concat toolbar--id '-overflow-menu') menu--IsExpanded=toolbar-overflow-menu-example--IsExpanded
```
Comment (mcoker) [inline src/patternfly/components/Toolbar/toolbar-overflow-menu-example.hbs:4]:
```suggestion
      overflow-menu--id=(concat toolbar--id '-overflow-menu')
      menu--IsExpanded=toolbar-overflow-menu-example--IsExpanded
```

File: src/patternfly/components/OverflowMenu/overflow-menu.hbs [hunk 2]
```diff
@@ -1,11 +1,29 @@
-{{#> wrapper overflow-menu-button--aria-expanded="true"}}
 <div class="{{pfv}}overflow-menu{{#if overflow-menu--modifier}} {{overflow-menu--modifier}}{{/if}}"
-  {{#if overflow-menu--attribute}}
-    {{{overflow-menu--attribute}}}
+  {{ternary overflow-menu--id 'id="{{overflow-menu--id}}"' ''}}
+  {{ternary overflow-menu--attribute 'id="{{overflow-menu--id}}"' ''}}>
```
Comment (mcoker) [inline src/patternfly/components/OverflowMenu/overflow-menu.hbs:3]: ID's are broken

Screenshot 2024-04-24 at 7 35 51 PM

I'm assuming these 2 lines should be something like this instead?



```suggestion
  {{ternary overflow-menu--id 'id="' overflow-menu--id '"' ''}}
  {{ternary overflow-menu--attribute overflow-menu--attribute ''}}>
```

**Review by mcoker — CHANGES_REQUESTED** [review]
[backstop report](https://github.com/patternfly/patternfly/files/15102011/backstop-4.24.24.pdf)

I'm sure some of these are known/purposeful since it's a WIP but since I've reviewed, IMO everything looks good  to merge except:
* The HBS helpers have console.logs that need to be removed
* The "New" menu example should be removed
* Overflow menu id's are broken, left a comment below on overflow.hbs

I'd like to merge this if possible so these examples are fixed. Currently they're throwing off the layout of lots of components (this visual regression report has 148 changes as result of this PR) and causing lots of issues with our screenshots. It would be great to fix those. If there is a different way of doing the hbs to make it more streamlined, that could always be done later and the visual regression tool can validate since it should(?) result in no visual changes. wdyt @srambach?

File: src/patternfly/components/NotificationDrawer/notification-drawer.scss [hunk 3]
```diff
@@ -1,5 +1,8 @@
 @use '../../sass-utilities' as *;
 
+// TODO: update alignment for text and action
+// TODO: update to use gap instead of margin
+
```
[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/NotificationDrawer/notification-drawer.scss:5]: https://github.com/patternfly/patternfly/issues/6577

**Review by mcoker — CHANGES_REQUESTED** [review]

File: src/patternfly/components/Menu/menu.scss [hunk 4]
```diff
@@ -4,7 +4,8 @@
   // * Menu
   --#{$menu}--RowGap: var(--pf-t--global--spacer--sm);
   --#{$menu}--Width: auto;
-  --#{$menu}--MinWidth: auto;
+  --#{$menu}--MinWidth: max-content;
+  --#{$menu}--MaxWidth: 400ch;
```
[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Menu/menu.scss:8]: Menu has no `max-width` control. 

Screenshot 2024-04-26 at 8 36 11 AM

**Comment by mattnolting** [comment]
[BackstopJS Report.pdf](https://github.com/patternfly/patternfly/files/15203693/BackstopJS.Report.pdf)

**Comment by srambach** [comment]
OK - here's the report I just ran https://drive.google.com/file/d/1sIz7YyVqInrF6f8M2gjFTjjsjO_1_aR7/view?usp=sharing
The only significant differences I see are 
- Overflow menu doesn't go into the kebab on mobile http://localhost:5000/components/table/html/overflow-menu-usage-desktop/
- Selected filters lost icons http://localhost:5000/components/toolbar/html/selected-filters-on-desktop-not-responsive/
- Toolbar lost its icons http://localhost:5000/components/toolbar/html/stacked-on-desktop/

**Review by mcoker — APPROVED** [review]
LGTM! 🚀

**Review by srambach — APPROVED** [review]
🏇

**Pull request merged**